packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js

```diff
@@ -30,6 +32,14 @@ const { ProgressBar } = unlock( componentsPrivateApis );
 function UploadFonts() {
 	const { installFont, notice, setNotice } = useContext( FontLibraryContext );
 	const [ isUploading, setIsUploading ] = useState( false );
+	const canUserCreate = useSelect( ( select ) => {
+		const { canUser } = select( coreStore );
+		return canUser( 'create', 'font-families' );
+	}, [] );
+
+	if ( ! canUserCreate ) {
+		return null;
+	}
```

Comment:
Given that we will render the upload tab only if `canUserCreate`, is this needed?

Reply:
Removed here: https://github.com/WordPress/gutenberg/pull/59332/commits/29d01c42c0724bcbe617534b177fd928a3f7e31a

Reply:
Sorry for the late response. I added the additional check based on the concern expressed here about the component being reused elsewhere and missing the check:

https://github.com/WordPress/gutenberg/pull/59294#discussion_r1500098717

Reply:
If the component is gonna be re-used, it can be added later. It probably will require extra work. I think that is out of the scope of this PR.